src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,122 @@
-public class StringManipulation implements StringManipulationInterface {
+import java.util.Arrays;
 
+public class StringManipulation implements StringManipulationInterface {
+	
+	private String[] wordArray;
+	
     @Override
     public String getString() {
-        return null;
+    	if (wordArray == null) {
+    		throw new NullPointerException("String has not been set for current StringManipulation object (hint: use setString()).");
+    	}
+        return String.join(" ", wordArray).trim();
     }
 
     @Override
     public void setString(String string) {
+    	if (string.isBlank()) {
+    		wordArray = new String[] { string };
+    	}
+    	else {
+    		wordArray = string.split(" ");
+    	}
     }
 
     @Override
     public int count() {
-        return 0;
+    	if (wordArray == null) {
+    		return -1;
+    	}
+    	if (getString().isBlank()) {
+    		return 0;
+    	}
+        return wordArray.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+    	String retVal = getString();
```

Comment:
You have to convert to string here to do your manipulations which means code is somewhat coupled and dependent on itself.